kubevirt_scripts/mountcd.sh

```diff
@@ -0,0 +1,220 @@
+#!/bin/bash
+set -ux -o pipefail
+
+#### IMPORTANT: This script is only meant to show how to implement required scripts to make custom hardware compatible with FakeFish.
+#### This script has to mount the iso in the server's virtualmedia and return 0 if operation succeeded, 1 otherwise
+#### Note: Iso image to mount will be received as the first argument ($1)
+#### You will get the following vars as environment vars
+#### BMC_ENDPOINT - Has the BMC IP
+#### BMC_USERNAME - Has the username configured in the BMH/InstallConfig and that is used to access BMC_ENDPOINT
+#### BMC_PASSWORD - Has the password configured in the BMH/InstallConfig and that is used to access BMC_ENDPOINT
+
+ISO=${1}
+CLUSTER_STORAGE_CLASS=ocs-storagecluster-ceph-rbd
+IS_HTTPS=false
+
+PVC_SPEC=$(cat <<EOF
+  spec:
+    accessModes:
+    - ReadWriteOnce
+    resources:
+      requests:
+        storage: 5Gi
+    storageClassName: ${CLUSTER_STORAGE_CLASS}
+EOF
+)
+
+if echo ${ISO} | grep -q https://; then
+  HOSTNAME=$(echo ${ISO} | awk -F "https://" '{print $2}' | awk -F "/" '{print $1}' | awk -F ":" '{print $1}')
+  PORT=$(echo ${ISO} | awk -F "https://" '{print $2}' | awk -F "/" '{print $1}' | awk -F ":" '{print $2}')
+  if [ -z ${PORT} ]; then
+    PORT=443
+  fi
+  openssl s_client -showcerts -connect ${HOSTNAME}:${PORT} </dev/null 2>/dev/null | openssl x509 -outform PEM > /tmp/iso-endpoint-ca.crt
+  if [ $? -eq 1 ]; then
```

Comment:
(Applies everywhere) you assume that the failure code is always 1. Strictly speaking, it can be any non-zero integer. If you prefer this approach (I still prefer `-e`), you need to use `-ne 0` or something like that.

Reply:
I prefer to stick to the current implementation. I changed -eq 1 to -ne 0 as you suggested. Thanks!.